trade-module/src/main/java/io/edpn/backend/trade/application/mappers/v1/FindCommodityDTOMapper.java

```diff
@@ -0,0 +1,35 @@
+package io.edpn.backend.trade.application.mappers.v1;
+
+import io.edpn.backend.trade.application.dto.v1.FindCommodityRequest;
+import io.edpn.backend.trade.application.dto.v1.FindCommodityResponse;
+import io.edpn.backend.trade.domain.filter.v1.FindCommodityFilter;
+import io.edpn.backend.trade.domain.model.CommodityType;
+import io.edpn.backend.trade.domain.model.ValidatedCommodity;
+
+import java.util.Optional;
+
+public class FindCommodityDTOMapper {
+    
+    public FindCommodityResponse map(ValidatedCommodity validatedCommodity) {
+        return FindCommodityResponse.builder()
+                .id(validatedCommodity.getId())
+                .commodityName(validatedCommodity.getCommodityName())
+                .displayName(validatedCommodity.getDisplayName())
+                .type(validatedCommodity.getType().toString())
+                .isRare(validatedCommodity.getIsRare())
+                .build();
+    }
+    
+    public FindCommodityFilter map(FindCommodityRequest findCommodityRequest) {
+        if (findCommodityRequest.getType() == null) {
```

Comment:
if can be replace with optional mapping

```suggestion
          return FindCommodityFilter.builder()
                .type(Optional.ofNullable(findCommodityRequest.getType()).map(CommodityType::valueOf).orElse(null))
                .isRare(findCommodityRequest.getIsRare())
                .build();
```